core/src/main/java/io/questdb/cairo/mv/MatViewDefinition.java

```diff
@@ -62,6 +64,8 @@ public class MatViewDefinition implements Mutable {
     public static final int REFRESH_TYPE_TIMER = 1;
     private static final Log LOG = LogFactory.getLog(MatViewDefinition.class);
     private String baseTableName;
+    private TimestampDriver baseTableTimestampDriver;
```

Comment:
Views survive base table being dropped and they will try to validate themselves when table is created again. What we need to seriously test for is a situation where view was created against one timestamp type and is being revalidated against another

Reply:
Yes, This is also why `updateBaseTableTimestampType` exists. We need to add tests for this.

Reply:
As discussed earlier, the MatView becomes invalid when the query's `timestampType` is inconsistent with the `timestampType` in MatVies's definition